### ISSUE TITLE
Standardize parameter-name field

### ISSUE THREAD

**Issue by Teddy-1000** [EURODEO/e-soh#21]
## The issue
There is not room for multiple and custom parameter-names in a query to an EDR api. Thus we need to standardize how we format acceptable names in this field so data from all providers are searchable under the same rules.

There are mainly four attributes needed to uniquely identify a type of measurement:
- standard_name
  - CF convention standard names from [this table](https://cfconventions.org/Data/cf-standard-names/current/build/cf-standard-name-table.html)
- level
  - meters above ground level the observation is taken.
- function
  - Many measurements are not instantaneous, but rather as an average, max, min or other. Available methods are defined by [CF Cell Methods](https://cfconventions.org/cf-conventions/cf-conventions.html#appendix-cell-methods)
- period
  - ISO8601 duration If the measurement is not instantaneous, they have been gathered over a set period. i.e 10 minutes 1 hours etc.

## Possible solution
I propose that we create a system where we combine the fields mentioned above as they are required for observations in E-SOH.These will be able to uniquely describe each type of  observation without introducing a new standard, but rather reuse standard fields already known. These should probably all be generated in ingest, and kept in the time series metadata.
```
standard_name_level_function_period
```
## Example of parameter names
````
air_temperature_2.0_max_PT10M
air_pressure_5.0_mean_PT1H
````
## Example of description of a parameter-name
Since the parameter-name is composite and not in its entirety part of a vocabulary, there will be need to describe it in a consistent way. Here I show one way of describing such a parameter-name.
```   
{
  "air_temperature_2.0m_max_PT10M": {
    "air_temperature": {
      "name": "standrd_name",
      "standard": "CF 1.4",
      "vocab": {
        "url": "https://vocab.nerc.ac.uk/collection/P07/current/CFSN0023/"
      }
    },
    "2.0m": {
      "name": "level",
      "explanation": "Meter from ground level of station"
    },
    "max": {
      "name": "function",
      "explanation": "What function is used on data collected over the given period"
    },
    "PT10M": {
      "name": "period",
      "explanation": "ISO8601 standard duration"
    }
  }
}
```

**Comment by Teddy-1000** [comment]
I have changed the issue a bit, going from a : seperator to _, this will also mean that leaving a field blank will not be possible.

**Comment by Teddy-1000** [comment]
After a few conversations and discussions, I believe it would be best to drop the functionality for a parsable and functional parameter-name, and rather treat parameter-name simply as a key, associated with each timeseries. This "key" will still be as described above, but will be generated and set by the ingestor based on mandatory fields as described above. Generating them in this way still leaves the option for a 3rd party client to parse and search in the available fields independently.
I have asked the datastore team to implement the necessary fields in the datastore and protobuf for this functionality.

**Comment by jo-asplin-met-no** [comment]
Proposal made [here](https://github.com/EURODEO/e-soh/pull/32).

**Comment by jo-asplin-met-no** [comment]
Note that the parameter name will actually not be a key in the strict sense: Two time series may share a given parameter name and e.g. differ in the _instrument_ field (example use case: two temperature sensors in the same location, one official and one experimental).

**Comment by Teddy-1000** [comment]
Closed by #44
